### PR TITLE
Jacobian trace approximation for weighted average distance metric.

### PR DESCRIPTION
**Pull request by ofirgo**
Implementing Jacobian trace approximation to be used for distance meאric weighted average.
This new method replaces the Hessian trace approximation as the model gradient feature for computing gradient-based weights.

### CODE REVIEW

File: model_compression_toolkit/core/common/framework_implementation.py [hunk 1]
```diff
@@ -382,8 +383,9 @@ def model_grad(self,
             alpha: A tuning parameter to allow calibration between the contribution of the output feature maps returned
                 weights and the other feature maps weights (since the gradient of the output layers does not provide a
                 compatible weight for the distance metric computation).
+            n_iter: The number of random iterations to calculate the approximated jacobian trace for each interest point.
 
-        Returns: A list of normalized gradients to be considered as the relevancy that each interest
+        Returns: A list of normalized jacobians traces to be considered as the relevancy that each interest
```
Comment (haihabi) [inline model_compression_toolkit/core/common/framework_implementation.py:388]: We are computing the trace(jacobians jacobians^T) and not the jacobians traces

File: model_compression_toolkit/core/pytorch/back2framework/model_gradients.py [hunk 2]
```diff
@@ -105,7 +106,8 @@ class PytorchModelGradients(torch.nn.Module):
     """
     def __init__(self,
                  graph_float: common.Graph,
-                 interest_points: List[BaseNode]):
+                 interest_points: List[BaseNode],
+                 output_list: List[BaseNode]):
         """
         Construct a PytorchModelGradients model.
```
Comment (haihabi) [inline model_compression_toolkit/core/pytorch/back2framework/model_gradients.py:113]: Missing argument in comments

File: model_compression_toolkit/core/common/framework_implementation.py (hunk 1 above)
Comment (ofirgo) [reply model_compression_toolkit/core/common/framework_implementation.py:388]: Is the term "jacobian-based weights" sounds ok? It's not possible to mention  trace(jacobians jacobians^T) everywhere in the comments (it's not just here, and it wouldn't be clear)
